charts/jenkins/templates/_helpers.tpl

```diff
@@ -48,6 +48,29 @@ If release name contains chart name it will be used as a full name.
 {{- end -}}
 {{- end -}}
 
+{{/*
+Returns the admin password
+https://github.com/helm/charts/issues/5167#issuecomment-619137759
+*/}}
+{{- define "jenkins.password" -}}
+  {{ if .Values.controller.adminPassword -}}
+    {{- .Values.controller.adminPassword | b64enc | quote }}
+  {{- else -}}
+    {{- $secret := (lookup "v1" "Secret" .Release.Namespace (include "jenkins.fullname" .)).data -}}
+    {{- if $secret -}}
+      {{/*
+        Reusing current password since secret exists
+      */}}
+      {{- index $secret ( .Values.controller.admin.passwordKey | default "jenkins-admin-password" ) -}}
```

Comment:
This is not only handy, but also needed as we mount secrets using `subPath`. Kubernetes does not support updating those. So for a new password to take effect a user would need to restart Jenkins.